test/unit/jsTest.js

```diff
@@ -0,0 +1,234 @@
+      fs.writeFileSync(pathsOfStaticJS[x], staticJSFiles[x])
+    }
+  }
+
+  afterEach(function (done) {
+    cleanupTestApp(appDir, (err) => {
+      if (err) {
+        throw err
+      } else {
+        done()
+      }
+    })
+  })
+
+  it('should compile all static js files using roosevelt-uglify', function (done) {
+    generateStaticFolder()
+    // create and init the app
+    generateTestApp({
+      appDir: appDir,
+      generateFolderStructure: true,
+      suppressLogs: {
+        httpLogs: true,
+        rooseveltLogs: true,
+        rooseveltWarnings: true
+      },
+      js: {
+        compiler: {
+          nodeModule: 'roosevelt-uglify',
+          showWarnings: false,
+          params: {}
+        },
+        whitelist: ['a.js', 'b.js', 'c.js']
+      }
+    }, 'initServer')
+    // create a fork of it and run it
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prod'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', () => {
+      // look into the .build folder to see if all the files were compiled and if there is any extras
+      const compiledJS = path.join(path.join(appDir, 'statics', '.build', 'js'))
+      const compiledJSArray = klawSync(compiledJS)
+      compiledJSArray.forEach((file) => {
+        let test = pathsOfCompiledJS.includes(file.path)
+        assert.equal(test, true)
+      })
+      testApp.kill()
+      done()
+    })
+  })
+
+  it('should only compile files that are whitelisted', function (done) {
+    generateStaticFolder()
+    // make a new array to compare the test files too
+    let pathOfWhiteListedFiles = [
+      path.join(appDir, 'statics', '.build', 'js', 'a.js'),
+      path.join(appDir, 'statics', '.build', 'js', 'c.js')
+    ]
+    // create and init the app
+    generateTestApp({
+      appDir: appDir,
+      generateFolderStructure: true,
+      suppressLogs: {
+        httpLogs: true,
+        rooseveltLogs: true,
+        rooseveltWarnings: true
+      },
+      js: {
+        compiler: {
+          nodeModule: 'roosevelt-uglify',
+          showWarnings: false,
+          params: {}
+        },
+        output: '.build/js',
+        sourceDir: 'js',
+        whitelist: ['a.js', 'c.js']
+      }
+    }, 'initServer')
+    // create a fork of it and run it
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prod'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', () => {
+      // test to see that only the whitelisted file was compiled
+      const compiledJS = path.join(path.join(appDir, 'statics', '.build', 'js'))
+      const compiledJSArray = klawSync(compiledJS)
+      compiledJSArray.forEach((file) => {
+        let test = pathOfWhiteListedFiles.includes(file.path)
+        assert.equal(test, true)
+      })
+      testApp.kill()
+      done()
+    })
+  })
+
+  it.skip('should minify all files except for those that are blacklisted', function (done) {
+    // change the data that would be put into the static files
```

Comment:
Why is this data being changed in this test?

Reply:
Oh yea for the blacklisted tests, you said that if the js files were too simple, their could be a chance that the minified version and the regular version would look the same, so for that test, I made each file a little bit more compicated

Reply:
No reason to not use the more complex js test files for every test